src/silx/gui/plot/CompareImages.py

```diff
@@ -927,8 +931,12 @@ def __updateData(self, updateColormap):
         vertical/horizontal separators moves.
         """
         raw1, raw2 = self.__raw1, self.__raw2
-        if raw1 is None or raw2 is None:
-            return
+        if raw1 is None:
+            raw1 = numpy.empty((1, 1))
+            raw1[0, 0] = numpy.nan
+        if raw2 is None:
+            raw2 = numpy.empty((1, 1))
+            raw2[0, 0] = numpy.nan
```

Comment:
This modification seems to make some stuff failing on my side. Because `setVisualizationMode` is called and this triggers `self.__updateColormap` which get issues because of data1 and data2 values.

But I don't really see why raw1 and raw2 become 2D numpy arrays with a nan ?

Reply:
The aim is to allow to compare a set image1 with an unset image2 (in case image2 is not yet set or a problem occured during the loading of image2)

So the user can see that one or the other image is missing.

Previously the previous state of the image2 was still displayed.

To make it easy, i have just replaced the None with a value which behave like an normal image (with NaN to make sure it is not taken by the user as a real image), so i know that any further processing will work.

But we can take a look at it together if you like.